qiskit_experiments/library/characterization/analysis/correlated_readout_error_analysis.py

```diff
@@ -0,0 +1,127 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Analysis class to characterize correlated readout error
+"""
+from typing import List, Tuple
+import numpy as np
+import matplotlib.pyplot as plt
+from qiskit.result import CorrelatedReadoutMitigator
+from qiskit_experiments.framework import ExperimentData
+from qiskit_experiments.framework.matplotlib import get_non_gui_ax
+from qiskit_experiments.framework import BaseAnalysis, AnalysisResultData, Options
+
+
+class CorrelatedReadoutErrorAnalysis(BaseAnalysis):
+    r"""
+    Correlated readout error characterization analysis
+
+    # section: overview
+
+        This class generates the full assignment matrix :math:`A` characterizing the
+        readout error for the given qubits from the experiment results.
+
+        :math:`A` is a :math:`2^n\times 2^n` matrix :math:`A` such that :math:`A_{y,x}`
+        is the probability to observe :math:`y` given the true outcome should be :math:`x`.
+
+        In the experiment, for each :math:`x`a circuit is constructed whose expected
+        outcome is :math:`x`. From the observed results on the circuit, the probability for
+        each :math:`y` is determined, and :math:`A_{y,x}` is set accordingly.
+
+        Returns
+
+            * The `Correlated readout error mitigator
+              <https://qiskit.org/documentation/stubs/qiskit.result.CorrelatedReadoutMitigator.html>`_
+              object (the assignment matrix can be accessed via its ``assignment_matrix()`` method).
+
+            * (Optional) A figure of the assignment matrix.
+
+
+    # section: reference
+        .. ref_arxiv:: 1 2006.14044
+    """
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        """Return default analysis options.
+
+        Analysis Options:
+            plot (bool): Set ``True`` to create figure for fit result.
+            ax (AxesSubplot): Optional. A matplotlib axis object to draw.
+        """
+        options = super()._default_options()
+        # since the plot size grows exponentially with the number of qubits, plotting is off by default
```

Comment:
Not in correlated, but yes in local (since local can handle many more qubits due to never generating the assignment matrix explicitly). I'll change this for correlated.